enos/provider/vmong5k.py

```diff
@@ -0,0 +1,80 @@
+import copy
+import logging
+
+from enoslib import api
+from enoslib.infra.enos_vmong5k.configuration import Configuration
+from enoslib.infra.enos_vmong5k.provider import VMonG5k as VMonG5K
+
+from . import g5k
+from .provider import Provider
+from ..utils import extra
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_CONFIG = {
+    'job_name': 'enos-vmong5k',
+    'walltime': '02:00:00'
+}
+
+
+def _build_enoslib_configuration(configuration):
+    _configuration = copy.deepcopy(configuration)
+    enoslib_configuration = _configuration.get("provider", {})
+    enoslib_configuration.pop("type", None)
+    if enoslib_configuration.get("resources") is not None:
+        return enoslib_configuration
+
+    machines = []
+    clusters = set()
+    resources = _configuration.get("resources", {})
+    resources = _configuration.get("topology", resources)
+    for description in extra.gen_enoslib_roles(resources):
+        for group in api.expand_groups(description["group"]):
+            clusters.add(description["flavor"])
+            machine = {
+                "roles": [group, description["role"]],
+                "nodes": description["number"],
+                "cluster": description["flavor"],
+                "min": 1
```

Comment:
We need to send a `flavour` here.
https://gitlab.inria.fr/discovery/enoslib/blob/master/enoslib/infra/enos_vmong5k/schema.py#L38-41

this depends on the role -- 
I'd say we can choose:
```
control: extra-large
network: large
compute: medium
# anything else: medium
```

available (defauklt flavour): https://gitlab.inria.fr/discovery/enoslib/blob/master/enoslib/infra/enos_vmong5k/constants.py

Reply:
I also realised that is not strictly required because I could test without defining the flavours.

Reply:
1. `nodes` and `min` shouldn't be accepted in the schema (this is a bug in enoslib[1])
2. same as above
3. one should use the advanced description by using a `resources` key under the `provider` key

[1]: https://gitlab.inria.fr/discovery/enoslib/issues/2